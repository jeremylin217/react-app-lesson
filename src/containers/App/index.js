import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import logo from '@/static/images/logo.svg'
import './style.css'

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <div className="mb-2">
            <Link className="btn btn-primary mr-1" to="/welcome">
              Welcome
            </Link>
            <Link className="btn btn-secondary mr-1" to="/example/redux">
              Counter Example
            </Link>
            <Link className="btn btn-secondary mr-1" to="/example/style">
              Styles Example
            </Link>
            <Link className="btn btn-secondary mr-1" to="/users">
              API Call Example
            </Link>
          </div>
          <p>
            Edit <code>src/containers/App/index.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    )
  }
}

export default App
