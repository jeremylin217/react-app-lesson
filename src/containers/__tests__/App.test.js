import React from 'react'
import toJson from 'enzyme-to-json'
import { render } from 'enzyme'
import App from '@/containers/App'
import { MemoryRouter } from 'react-router'

describe('<App />', () => {
  const wrapper = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  )

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should include a link to /welcome', () => {
    const link = wrapper.find('.btn.btn-primary').first()
    expect(link.text()).toEqual('Welcome')
    expect(link.prop('href')).toEqual('/welcome')
  })
})
