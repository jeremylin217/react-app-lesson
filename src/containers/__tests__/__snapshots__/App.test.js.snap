// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <img
      alt="logo"
      class="App-logo"
      src="logo.svg"
    />
    <div
      class="mb-2"
    >
      <a
        class="btn btn-primary mr-1"
        href="/welcome"
      >
        Welcome
      </a>
      <a
        class="btn btn-secondary mr-1"
        href="/example/redux"
      >
        Counter Example
      </a>
      <a
        class="btn btn-secondary mr-1"
        href="/example/style"
      >
        Styles Example
      </a>
      <a
        class="btn btn-secondary mr-1"
        href="/users"
      >
        API Call Example
      </a>
    </div>
    <p>
      Edit 
      <code>
        src/containers/App/index.js
      </code>
       and save to reload.
    </p>
    <a
      class="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
